index.js

```diff
@@ -89,13 +89,28 @@ function create(win, opts) {
 
 		// filter out leading/trailing separators
 		// TODO: https://github.com/electron/electron/issues/5869
-		menuTpl = menuTpl.filter((el, i, arr) => !(el.type === 'separator' && (i === 0 || i === arr.length - 1 || arr[i + 1].type === 'separator')));
+		menuTpl = deleteUnnecessarySeparatorsAndElements(menuTpl);
 
-		const menu = (electron.Menu || electron.remote.Menu).buildFromTemplate(menuTpl);
-		menu.popup(win);
+		if (menuTpl.some(elem => isVisible(elem))) {
+			const menu = (electron.Menu || electron.remote.Menu).buildFromTemplate(menuTpl);
+			menu.popup(win);
+		}
+	});
+}
+
+function deleteUnnecessarySeparatorsAndElements(menuTpl) {
+	let visiblePreviousEl;
+	return menuTpl.filter(el => isVisible(el)).filter((el, i, arr) => {
+		const toDelete = el.type === 'separator' && (!visiblePreviousEl || i === arr.length - 1 || arr[i + 1].type === 'separator');
+		visiblePreviousEl = toDelete || !isVisible(el) ? visiblePreviousEl : el;
```

Comment:
Why are you doing `!isVisible(el)` check? You already filtered out all invisible elements on line 103.